stores/multipart.go

```diff
@@ -181,8 +181,8 @@ func (s *SQLStore) MultipartUploadParts(ctx context.Context, bucket, object stri
 		}
 		// Check if there are more parts beyond 'limit'.
 		if limitUsed && len(dbParts) == int(limit) {
-			resp.IsTruncated = true
-			resp.NextMarker = dbParts[len(dbParts)-1].PartNumber
+			resp.HasMore = true
+			resp.NextMarker = dbParts[len(dbParts)-2].PartNumber
```

Comment:
this seems wrong? edit: also I think we could handle a `limit` of 0 better

Reply:
Well if the limit is 1 then we fetch 2. Then we want the marker to be the one at index 0 which is Len-2.
Because otherwise we are going to skip an element on the next call.

Define "better"?

Reply:
this is a `nit` but can we move `dbParts = dbParts[:len(dbParts)-1]` up 2 lines and then set `NextMarker` to be the last element

Reply:
gets rid of the weird `-2` 🤷‍♂️